api/errors/connectrpc.go

```diff
@@ -78,3 +84,72 @@ func NewHelpLink(description, url string) *errdetails.Help_Link {
 		Url:         url,
 	}
 }
+
+// NewExternalErrorDetail can be called to transform an internal error into an
+// external error that is safe to be returned in gRPC/Connect.
+// Error messages are removed.
+func NewExternalErrorDetail(message string, details ...proto.Message) *commonv1.ExternalError {
+	e := commonv1.ExternalError{
+		Message: message,
+	}
+
+	for _, detail := range details {
+		anyDetail, err := anypb.New(detail)
+		if err != nil {
+			// Ignore errors
+			continue
+		}
+		e.Details = append(e.Details, anyDetail)
+	}
+
+	return &e
+}
+
+// NewSafePublicError can be called to transform an internal error into an
+// external error that is safe to be returned in gRPC/Connect.
+// Error messages are removed.
+func NewSafePublicError(internalError *status.Status) *status.Status {
```

Comment:
I wonder: Is that name accurate given that it retries and returns type `status.Status`?

Reply:
I am just keeping the same name that was used elsewhere.